### DISCUSSION TITLE
What would cause the feed to not show random posts which are actually in the database?

### DISCUSSION THREAD

**Discussion by venteto** [MarshalX/atproto#254] [Questions] [answered]
This one of my [test feeds](https://bsky.app/profile/did:plc:wecucphzbhaznb64qmvto5oj/feed/museums).
This is the [curated user list](https://bsky.app/profile/did:plc:wecucphzbhaznb64qmvto5oj/lists/3kjvtzbraol2i) it is being compared to.
This is [the repo](https://github.com/venteto/bfg2) for the feed generator.

When I look at the feed and notice some of the posts from the user list are not showing up, I ssh into the remote container, and look in the SQLite database. I see posts that are not showing up in the feed did actually get saved, from the firehose indexer, into the database, but I can't imagine why random posts in the database would not display.

Here's a sample of what the database looks like, including posts that are not showing up in the feed (id's 5, 12, 14 and 17):

![Screenshot from 2024-01-26 19-13-50](https://github.com/MarshalX/atproto/assets/7062385/08d2d63f-a9ed-4942-a3a8-d140dc937fca)

Thank you for any insight.

**Comment by MarshalX** [comment]
I can recommend setting a breakpoint in the Flask app and debugging it locally. You can just send requests to your server without bsky at all 

From the code perspective, I can recommend `.where()` right after `.select()` at https://github.com/venteto/bfg2/blob/601fc670d84f2c916be7ece62847b4be8ff3c6b3/responder/generators/museums.py#L12-L13. but i don't know depth of peewee

**Reply by venteto** [reply]
I moved `.where()` as you suggested, and I also tried disabling the `.limit()` for a bit but neither change seems to make any difference. I don't know Python well enough to how to debug it with breakpoints. But I tried placing 

```python
import pdb; pdb.set_trace()
```

after [https://github.com/venteto/bfg2/blob/main/responder/generators/museums.py#L17](https://github.com/venteto/bfg2/blob/main/responder/generators/museums.py#L17) and running Flask locally. The output didn't mean anything to me, but I don't know what I'm doing either.

I also tried looking at the generated SQL as [described here](https://julianrawcliffe.wordpress.com/2019/09/15/python-flask-peewee-sql-debugging/) while running Flask locally (which initially made me think I should remove the `.limit()` clause ... because at first that _did_ seem to actually fix the problem but now it seems like that was just random luck before redeploying the app with a fresh, empty database.

I did also notice something that seems odd to me, on both localhost (after `rsync`ing the live database back to my laptop) and on the remote server. An example of a post which does not display on the live feed, and is actually in the database ... also shows up in the JSON response from the actual Flask app ... is it possible the issue is really with the `bsky.app` client that is requesting the JSON from my Fly.io server?

Remote server:
![MUSEUM-missing-post-is-in-response-on-FLY](https://github.com/MarshalX/atproto/assets/7062385/8bd12d11-b69f-47ce-877a-71757d1a2e51)

Localhost:
![MUSEUM-missing-post-is-in-response-on-LOCAL](https://github.com/MarshalX/atproto/assets/7062385/284467ff-935c-45c7-922d-1e9aaf91be44)

The highlighted row in the JSON responses [is this painting](https://bsky.app/profile/moma.bsky.social/post/3kjyfg67jta2j), which shows up in my List, but not in my Feed when viewed on `bsky.app`
https://bsky.app/profile/moma.bsky.social/post/3kjyfg67jta2j

**Reply by MarshalX** [reply]
Nice catch with json! If the Flask app returns it correctly and the URI is valid (you can get a post using this URI) it means that the problem is not on your side at all

**Reply by venteto** [reply]
I opened [an issue in](https://github.com/bluesky-social/social-app/issues/2657) Bluesky's social app repo

**Reply by MarshalX** [reply, marked as answer]
got the answer from the issue above

tl;dr  you enabled the language filter for your account at  https://bsky.app/settings/language. that's why not all posts show up. deselect all languages from the "Content Languages" sections

**Reply by venteto** [reply]
I am closing this for now based on this

I don't even recall touching that setting in the app but I'm glad I know the problem now :)
